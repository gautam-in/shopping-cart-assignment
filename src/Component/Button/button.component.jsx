import React from "react";
import "./button.styles.scss";

function Button({ text, onClick = null }) {
  return (
    <div className="button-cont">
      <button className="button-prm" onClick={() => onClick && onClick()} aria-label={text}>
        {text}
      </button>
    </div>
  );
}

export default Button;
