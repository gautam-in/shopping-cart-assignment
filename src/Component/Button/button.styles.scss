.button-cont {
  display: flex;
  justify-content: center;
  .button-prm {
    background-color: rgb(211, 62, 62);
    padding: 1vh;
    font-size: medium;
    min-width: 10vw;
    color: #fff;
    border: none;
    border-radius: 5px;
  }
  .button-prm:active {
    background-color: rgb(231, 81, 81);
  }
}

@media (max-width: 425px) {
  .button-cont {
    .button-prm {
      font-size: 1vh;
    }
  }
}
