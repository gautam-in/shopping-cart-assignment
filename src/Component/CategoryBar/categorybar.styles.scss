.category-container {
  display: flex;
  margin-left: 5vw;
  margin-right: 5vw;
  box-shadow: 0px 8px 6px -6px;

  .categoryImage-container {
    display: flex;
    justify-content: center;
    width: 40vw;
    box-sizing: border-box;
    padding: 2vw;

    .category-image {
      width: 30vw;
      // height: 30vh;
    }
  }

  .categoryText-container {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 60vw;
    box-sizing: border-box;
    flex-direction: column;

    .category-title {
      text-align: center;
    }

    .category-button {
      display: flex;
      justify-content: center;
    }
  }
}

@media (max-width: 425px) {
  .category-container {
    padding: 2vh;
    .categoryImage-container {
      .category-image {
        width: 30vw;
        height: 10vh;
      }
    }

    .categoryText-container {
      font-size: 1vh;
    }
  }
}
