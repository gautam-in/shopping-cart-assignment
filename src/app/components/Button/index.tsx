/**
 *
 * Button
 *
 */
import React, { memo } from 'react';
import { Button as MuiButton, ButtonProps } from '@mui/material';
import { createTheme, ThemeProvider, styled } from '@mui/material/styles';

interface Props extends ButtonProps {
  label: string;
}

const theme = createTheme({
  palette: {
    primary: {
      main: '#F92A53',
    },
  },
});

export const Button = memo((props: Props) => {
  const { label, ...rest } = props;

  return (
    <ThemeProvider theme={theme}>
      <CustomButton variant="contained" {...rest}>
        {label}
      </CustomButton>
    </ThemeProvider>
  );
});

const CustomButton = styled(MuiButton)`
  width: 60%;
  border-radius: 0px;
  text-transform: capitalize;
`;
