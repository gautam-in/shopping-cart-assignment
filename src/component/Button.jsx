import "../styles/button.scss";

const BUTTON_TYPE_CLASSES = {
  max: "max",
  buy: "buy",
  checkout: "checkout",
};
const Button = ({ children, buttonType, ...otherProps }) => {
  return (
    <button
      className={`button-container ${BUTTON_TYPE_CLASSES[buttonType]}`}
      {...otherProps}
    >
      {children}
    </button>
  );
};

export default Button;
