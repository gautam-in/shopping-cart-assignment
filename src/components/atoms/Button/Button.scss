@import "../../../styles/variables";

// Default Class for Buttons
.button-wrapper {
  display: block;
  border-radius: 3px;
  text-align: center;
  text-decoration: none;
  font-size: $text-font-size;
  padding: 0.5rem;
  border: 1px solid #eee;
  background-color: $button-color;
  color: $button-text-color;
  cursor: pointer;
  position: relative;

  &:active {
    top: 0.6px;
    left: 0.6px;
  }

  &:disabled {
    background-color: $secondary-color;
    cursor: not-allowed;
  }

  @media screen and (max-width: $breakpoint-mobile) {
    font-size: $small-font-size;
    padding: 0.4rem;
  }
}
