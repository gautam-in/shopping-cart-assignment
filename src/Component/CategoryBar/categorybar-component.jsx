import React from "react";
import "./categorybar.styles.scss";
import Button from "../Button/button.component";

function CategoryBar({ imageURL, name, id, desc, className, onClick }) {
  return (
    <div className={className} key={id}>
      <div className="categoryImage-container">
        <img
          className="category-image"
          src={`${process.env.PUBLIC_URL}${imageURL}`}
          alt={name}
          key={id}
        />
      </div>
      <div className="categoryText-container">
        <h2 className="category-title">{name}</h2>
        <p>{desc}</p>
        <div className="category-button">
          <Button text={`Explore ${name}`} onClick={onClick} />
        </div>
      </div>
    </div>
  );
}

export default CategoryBar;
