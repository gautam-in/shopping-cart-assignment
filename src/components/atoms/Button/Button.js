import React from "react";
import "./Button.scss";

export default function Button({
  children,
  className = "",
  onClick,
  ...props
}) {
  return (
    <button
      className={`${className} button-wrapper`}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  );
}
