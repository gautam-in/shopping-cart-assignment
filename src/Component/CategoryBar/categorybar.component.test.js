import { shallow } from "enzyme";
import React from "react";
import Button from "../Button/button.component";
import CategoryBar from "./categorybar-component";

describe("expects to test the use cases of component", () => {
  const toBeCalled = jest.fn();
  const wrapper = shallow(
    <CategoryBar
      imageURL={"/static/image"}
      name="Fruits"
      id={"5a32ef"}
      className="conatiner"
      desc={"fruits"}
      onClick={toBeCalled}
    />
  );

  it("expects to render component with given props", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("expects to test props passed to component", () => {
    expect(wrapper.find("img").prop("alt")).toBe("Fruits");
    expect(wrapper.find("img").prop("src")).toBe(
      `${process.env.PUBLIC_URL}/static/image`
    );
    expect(wrapper.find("h2").prop("children")).toBe("Fruits");
    expect(wrapper.find("p").prop("children")).toBe("fruits");
    expect(wrapper.containsMatchingElement(<Button />)).toBe(true);
  });
});
