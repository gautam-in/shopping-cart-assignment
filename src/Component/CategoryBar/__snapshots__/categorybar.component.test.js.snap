// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to test the use cases of component expects to render component with given props 1`] = `ShallowWrapper {}`;
